Sources/Starknet/Accounts/StarknetAccount.swift

```diff
@@ -80,28 +113,67 @@ public class StarknetAccount: StarknetAccountProtocol {
         if let paramsMaxFee = params.maxFee {
             maxFee = paramsMaxFee
         } else {
-            let feeEstimate = try await estimateFee(calls: calls, nonce: nonce)
+            let feeEstimate = try await estimateFeeV1(calls: calls, nonce: nonce)
             maxFee = estimatedFeeToMaxFee(feeEstimate.overallFee)
         }
 
-        let signParams = StarknetExecutionParams(nonce: nonce, maxFee: maxFee)
-        let transaction = try sign(calls: calls, params: signParams, forFeeEstimation: false)
+        let signParams = StarknetDeprecatedExecutionParams(nonce: nonce, maxFee: maxFee)
+        let transaction = try signV1(calls: calls, params: signParams, forFeeEstimation: false)
 
         let result = try await provider.addInvokeTransaction(transaction)
 
         return result
     }
 
-    public func estimateFee(calls: [StarknetCall], nonce: Felt) async throws -> StarknetFeeEstimate {
-        let signParams = StarknetExecutionParams(nonce: nonce, maxFee: .zero)
-        let transaction = try sign(calls: calls, params: signParams, forFeeEstimation: true)
+    public func executeV3(calls: [StarknetCall], params: StarknetOptionalExecutionParamsV3) async throws -> StarknetInvokeTransactionResponse {
+        var nonce: Felt
+        var resourceBounds: StarknetResourceBoundsMapping
+
+        if let paramsNonce = params.nonce {
+            nonce = paramsNonce
+        } else {
+            nonce = try await getNonce()
+        }
+
+        if let paramsResourceBounds = params.resourceBounds {
+            resourceBounds = paramsResourceBounds
+        } else {
+            let feeEstimate = try await estimateFeeV3(calls: calls, nonce: nonce)
+            resourceBounds = feeEstimate.toResourceBounds()
+        }
```

Comment:
I think the logic with `StarknetOptionalExecutionParamsV3` is not ideal, and overloaded functions like in jvm would be better. 
But to not further complicate this PR, I decided to follow the approach already used for v1 txs.

Reply:
#136